Sources/CoreCommands/Options.swift

```diff
@@ -261,11 +261,12 @@ public struct BuildOptions: ParsableArguments {
                 shouldDisplay: false))
     public var xcbuildFlags: [String] = []
 
-    @Option(name: .customLong("Xmanifest", withSingleDash: true),
+    @Option(name: [.customLong("Xmanifest", withSingleDash: true),
+                   .customLong("Xhost", withSingleDash: true)],
             parsing: .unconditionalSingleValue,
-            help: ArgumentHelp("Pass flag to the manifest build invocation",
+            help: ArgumentHelp("Pass flag to build invocation targetting the host (manifest and plugins)",
```

Comment:
I'm not sure I like the idea of `-Xmanifest` being an alias here, that feels unexpected, especially if we are doing it in perpetuity. I think these should be separate with `-Xhost` affecting everything and `-Xmanifest` to continue to only affect manifests specifically and emitting a warning about it being deprecated. Then we can eventually remove `-Xmanifest` and the support for separate manifest flags.

Reply:
That seems like a good idea.  And if you specify both, then `-Xmanifest` wins, I'd assume.

Reply:
+1 on @neonichu idea.

also, how about `-Xhost-swiftc`? `-Xhost` seems a bit too generic, and this only apply to `swiftc` and this way we can also add `-Xhost-clang` etc

Reply:
This is a good point, but I think `-Xswiftc-host` might be a better spelling?

Reply:
I agree with @neonichu.  Being able to, in essence, apply `-host` as a suffix to any of the tools makes a lot of sense — each of them could really be used for the host in addition to the destination, so it's really another axis.  And I think a suffix probably makes things more readable than a prefix (it's a nitpick of course), but it flows better).

Reply:
> I'm not sure I like the idea of `-Xmanifest` being an alias here, that feels unexpected, especially if we are doing it in perpetuity. I think these should be separate with `-Xhost` affecting everything and `-Xmanifest` to continue to only affect manifests specifically and emitting a warning about it being deprecated. Then we can eventually remove `-Xmanifest` and the support for separate manifest flags.

I considered this approach but left it aside because I think as currently implemented will automatically work for people that used to use `-Xmanifest` and found out about plugins not getting the same flags applied to (in my mind `-Xmanifest` was always "flags that my host needs to compile, not my target". I was also hoping to have an easier upgrade path for our scripts (which are used with stable versions of SwiftPM and the `main` version of SwiftPM) until all of the versions supported `-Xhost` and then switch. I will see what it takes to do as you ask, but I wanted to see a different PoV.

> That seems like a good idea. And if you specify both, then `-Xmanifest` wins, I'd assume.

I imagine that you mean if any `-Xmanifest` is provided, only the manifest compilation will not use any of the `-Xhost` arguments? Not that `-Xmanifest` will be used even for plugins, right? I don't know if someone will need that behaviour explicitly, and I am a little bit afraid of implementing it, since someone will end up relying on it. I will give it a shot if this allow this to be merged (as I understand it as explained at the start of this paragraph).

> +1 on @neonichu idea.
> 
> also, how about `-Xhost-swiftc`? `-Xhost` seems a bit too generic, and this only apply to `swiftc` and this way we can also add `-Xhost-clang` etc

> This is a good point, but I think `-Xswiftc-host` might be a better spelling?

I think there's no `-Xmanifest-swiftc` and `-Xmanifest-clang` because `-Xmanifest` can be composed with other `-Xfoo` accepted by `swiftc`.  We use this to correctly pass the right arguments to each piece of the toolchain (example: `-Xmanifest -Xcc -Xmanifest -nostdlibinc -Xmanifest -Xclang-linker -Xmanifest --sysroot=/var/empty`). This might be because the manifest only invoked `swiftc`, and plugins only seem to do so as well. Is there any case you all think that products that need to be compiled for the host will need to invoke something else but `swiftc`? I will keep this suggestion out of the rework. If you really think this change is important, I will abide, but I really think it makes things more complicated than needed.

That say, if I have some vote in the spelling, I think `-Xhost-swiftc` is slightly better (having host first indicates earlier that this is not a normal `-Xswiftc`.

Reply:
great points @drodriguez, to add a few thoughts:

1. manifests are by definition swift only, but we should consider if plugins could include other languages in the future. if they do we should design the flags to support different compilers. 
2. I agree with your argument that `-Xhost-swiftc` is better.

@neonichu @abertelrud, thoughts?

Reply:
> Is there any case you all think that products that need to be compiled for the host will need to invoke something else but swiftc?

Yes, this was the motivation for suggesting to broaden the discussion here.  There are flags for other tools (`-Xcc`, `-Xlinker`, etc) and for each of those it's quite possible that the flags will need to be different for the host vs the destination during a regular build (as opposed to compilation of a manifest or plugin, which today is handled outside the build system).

I didn't realize that the compiler driver allows these flags to be composed, however.  It's important to support Clang as well, but if it also supports composition, then doing `-Xmanifest -Xlinker -Xmanifest -lfoo` (for example) seems like a way around that.

I still think that the name `-Xhost` is too generic in this case, if it only applies to compilation of manifests and plugins, since I'd expect that with that name it would also apply to products built for the host.

> I agree with your argument that -Xhost-swiftc is better.

This order seems fine.

Reply:
> I still think that the name -Xhost is too generic in this case, if it only applies to compilation of manifests and plugins, since I'd expect that with that name it would also apply to products built for the host.

I used the name because it is the name of the "destination" used for the manifest. I am fine changing it if someone can find a good common name that applies to both manifest and plugins.

If we follow compnerd recommendation (https://github.com/apple/swift-package-manager/pull/5966#issuecomment-1364714796) and want to follow the GCC naming (https://gcc.gnu.org/onlinedocs/gccint/Configure-Terms.html), one possibility would be `-Xbuild`, but that might be confusing with internal terminology (and the fact that the concepts of "target" and "host" are a little bit dilutted by LLVM architecture).

Finally, if we want to add this as `-X<whatever>-swiftc` in this PR I am fine doing that change. `-Xmanifest` will be left as a deprecated detail that did not support several tools.